runhouse/resources/resource.py

```diff
@@ -253,6 +253,64 @@ def str_dict_or_resource_to_str(val):
                 return None
         return config
 
+    @classmethod
+    def _update_system_in_config(cls, config):
```

Comment:
What you're doing here and in `update_creds_in_config` is already supported via Resource._check_for_child_configs. Take a look at how Module ensures that the system and env are properly reloaded from a config (or string) inside Module._check_for_child_configs. We shouldn't be doing this within the resource class because it is not presumed that a resource has a system or creds as a sub-resource. Every subclass can overload to specify what sub-resources they need to reconstruct from the config.